lib/holidays/date_calculator/weekend_modifier.rb

```diff
@@ -16,24 +16,47 @@ def to_monday_if_sunday(date)
         date
       end
 
+      # if Christmas falls on a Sunday, move it to the next Tuesday (Boxing Day will go on Monday)
+      # if Christmas falls on a Saturday, move it to the next Monday (Boxing Day will be Sunday and potentially Tuesday)
+      # used as a callback function, if xmas is not observed on the 25th
+      def xmas_to_weekday_if_weekend(year)
+        to_tuesday_if_sunday_or_monday_if_saturday(Date.civil(year, 12, 25))
+      end
+
       # Move Boxing Day if it falls on a weekend, leaving room for Christmas.
       # Used as a callback function.
       def to_weekday_if_boxing_weekend(date)
         if date.wday == 6 || date.wday == 0
           date += 2
-        elsif date.wday == 1
+        elsif date.wday == 1 # https://github.com/holidays/holidays/issues/27
           date += 1
         end
 
         date
       end
 
+      # if Christmas falls on a Saturday, move it to the next Monday (Boxing Day will be Sunday and potentially Tuesday)
+      # if Christmas falls on a Sunday, move it to the next Tuesday (Boxing Day will go on Monday)
+      #
+      # if Boxing Day falls on a Saturday, move it to the next Monday (Christmas will go on Friday)
+      # if Boxing Day falls on a Sunday, move it to the next Tuesday (Christmas will go on Saturday & Monday)
+      def to_tuesday_if_sunday_or_monday_if_saturday(date)
+        date += 2 if [0, 6].include?(date.wday)
+        date
+      end
+
       # Call to_weekday_if_boxing_weekend but first get date based on year
       # Used as a callback function.
-      def to_weekday_if_boxing_weekend_from_year(year)
+      def to_weekday_if_boxing_weekend_from_year_or_to_tuesday_if_monday(year)
```

Comment:
`to_weekday_if_boxing_weekend_from_year_or_to_tuesday_if_monday` calls `to_weekday_if_boxing_weekend`, while `to_weekday_if_boxing_weekend_from_year` calls `to_tuesday_if_sunday_or_monday_if_saturday`

`to_weekday_if_boxing_weekend` has a special case for if Boxing Date is on a Monday, where if you have Xmas on a Sunday and Boxing Day on a Monday, both are nudged forward a day - Xmas becomes Monday and Boxing Day becomes Tuesday (see https://github.com/holidays/holidays/issues/27)

`to_tuesday_if_sunday_or_monday_if_saturday` has no special case for Mondays, so in the scenario above, Boxing Day would stay on Monday, but Xmas would be observed on Tuesday. See http://publicholidays.com.au/qld

The reason for renaming was that the logic in `to_tuesday_if_sunday_or_monday_if_saturday` was newly added, and I was assuming all other uses would expect the `to_weekday_if_boxing_weekend` logic.